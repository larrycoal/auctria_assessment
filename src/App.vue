<template>
  <header class="flex p-2 text-xl text-center bg-slate-900 text-gray-50 nav">
    <h2 class="mx-2 flex flex-1 items-start font-bold">Ticketster</h2>
    <nav class="desktop_nav">
      <router-link class="mx-4" to="/"
        ><i class="fa-solid fa-house" style="color: #ffffff"></i
        >Home</router-link
      >
      <router-link class="mx-4" to="/admin"
        ><i class="fa-solid fa-screwdriver-wrench" style="color: #ffffff"></i
        >Admin</router-link
      >
      <router-link class="mx-4 position:relative" to="/cart"
        ><i class="fa-solid fa-cart-shopping" style="color: #ffffff"></i>Cart
        <span>{{ allCart.length }}</span></router-link
      >
    </nav>
    <div @click="handleShowNav" class="hamburger">
      <i class="fa-solid fa-bars" style="color: #f7f7f8"></i>
    </div>
    <nav v-if="shownav" class="mobile_nav">
      <router-link class="mx-4" to="/"
        ><i class="fa-solid fa-house" style="color: #ffffff"></i
        >Home</router-link
      >
      <router-link class="mx-4" to="/admin"
        ><i class="fa-solid fa-screwdriver-wrench" style="color: #ffffff"></i
        >Admin</router-link
      >
      <router-link class="mx-4 position:relative" to="/cart"
        ><i class="fa-solid fa-cart-shopping" style="color: #ffffff"></i>Cart
        <span>{{ allCart.length }}</span></router-link
      >
    </nav>
  </header>
  <router-view />
</template>

<script lang="ts">
import { defineComponent,ref } from "vue";
import { mapGetters } from "vuex";

export default defineComponent({
  setup(){
    let shownav = ref<boolean>(false)
    const handleShowNav = ()=>{
      shownav.value = !shownav.value
    }
    return{shownav,handleShowNav}
  },
  computed: {
    ...mapGetters(["allCart"]),
  },
});
</script>
<style scoped>
i {
  margin: 0 5px;
  height: 10px;
}
.nav{
  position: relative;
}
.nav a {
  display: flex;
  align-items: baseline;
  gap: 5px;
}
.mobile_nav{
  display: none;
}
.nav a span {
  color: red;
  font-size: 12px;
  align-self: flex-start;
  font-weight: bold;
  height: fit-content;
}
.hamburger {
  display: none;
}
.desktop_nav{
  display: flex;
}
@media screen and (max-width: 400px) {
  .nav .desktop_nav {
    display: none;
  }
  .hamburger {
    display: block;
  }
  .mobile_nav{
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 10px;
    padding: 10px;
    position: absolute;
    background: #0f172a;
    width: 100%;
    z-index: 3;
    top: 42px;
    left: 0;
  }
}
</style>