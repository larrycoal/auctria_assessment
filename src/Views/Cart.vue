<template>
  <h2>Your Cart</h2>
  <table>
    <tr>
      <th>Item</th>
      <th>Name</th>
      <th>Description</th>
      <th>Price</th>
    </tr>
    <tr v-for="(ticket, idx) in allCart" :key="idx">
      <td>
        <img
          src="https://res.cloudinary.com/dbdrtuscd/image/upload/v1701885193/Screenshot_2023-12-06_at_12.51.35_k2pgtu.png"
          alt="icon for a ticket"
        />
      </td>
      <td>{{ ticket.ticketName }}</td>
      <td>{{ ticket.description }}</td>
      <td>{{ ticket.price }}cad</td>
    </tr>
  </table>
  <div class="cart_bottom">
    <div class="container">
      <p><span>Amount:</span><span>{{amount}}cad</span></p>
      <p><span>HST:</span><span>10%</span></p>
      <p><span>Total:</span><span>{{total}}cad</span></p>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { mapGetters, useStore } from "vuex";
import eventType from "../types/event";
export default defineComponent({
  setup() {
    const store = useStore();
    let amount: number = 0;
    let total:number = 0
    store.state.cart.forEach((i: eventType) => {
      amount += i.price;
    });
    total = (amount * 0.1) + amount
    return { amount, total };
  },
  computed: {
    ...mapGetters(["allCart"]),
  },
});
</script>
<style scoped>
h2 {
  text-align: center;
  font-size: 3rem;
}
table {
  font-family: arial, sans-serif;
  border-collapse: collapse;
  width: 50%;
  margin: 2rem auto;
}

td,
th {
  border: 1px solid #dddddd;
  text-align: left;
  padding: 8px;
}
tr img {
  width: 150px;
}
.cart_bottom {
  width: 50%;
  margin: 1rem auto;
  display: flex;
  justify-content: flex-end;
}
.cart_bottom .container {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  gap: 10px;
  width: fit-content;
}
.cart_bottom .container p {
  display: flex;
  width: 100%;
  display: flex;
  justify-content: space-between;
  gap: 10px;
}
.cart_bottom .container p:nth-child(3) {
  border-top: 1px solid black;
  border-bottom: 1px solid black;
}
.cart_bottom .container p span:nth-child(1) {
  font-weight: bold;
}
@media screen and (max-width: 400px) {
  table {
    width: 90%;
    margin: 10px;
  }
  .cart_bottom{
    width: 90%;
  }
}
</style>

