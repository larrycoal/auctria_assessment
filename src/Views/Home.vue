<template>
  <div>
    <div class="hero">
      <h2>Welcome To Ticketster</h2>
      <p>Your one stop shop to all your ticket needs</p>
    </div>
    <h2 class="text-center text-4xl">Buy Tickets</h2>
    <ul class="tickets_wrapper" v-if="allTickets.length">
      <li class="card" v-for="(ticket, idx) in allTickets" :key="idx">
       <div v-if="!ticket.isVip">
         <img
          src="https://res.cloudinary.com/dbdrtuscd/image/upload/v1701885193/Screenshot_2023-12-06_at_12.51.35_k2pgtu.png"
          alt="icon for a ticket"
        />
       </div>
       <div v-if="ticket.isVip">
         <img
          src="https://res.cloudinary.com/dbdrtuscd/image/upload/v1701885193/Screenshot_2023-12-06_at_12.51.27_vkvvjv.png"
          alt="icon for a ticket"
        />
       </div>
        <h3>{{ ticket.ticketName }}</h3>
        <p>{{ ticket.description }}</p>
        <p class="sub">
          <span>{{ ticket.price }}cad</span
          ><span>Available:{{ ticket.count }}</span>
        </p>
        <button class="add_to_cart" v-if="ticket.isVip" v-on:click="handleAddToCart(ticket)">
          <span class="triangle"></span>
          Vip Ticket
          <span class="triangle"></span>
        </button>
        <button class="add_to_cart" v-if="!ticket.isVip" v-on:click="handleAddToCart(ticket)">
          <span class="triangle"></span>Regular Ticket<span
            class="triangle"
          ></span>
        </button>
      </li>
    </ul>
    <div class="text-5xl" v-else>
      <p class="text-center my-4">No ticket on sale presently</p>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import { mapGetters, useStore } from "vuex";
import eventType from "../types/event";

export default defineComponent({
  setup() {
    const store = useStore();
    // handle when user add ticket to cart
    const handleAddToCart = (ticket: eventType) => {
      store.commit("addToCart", ticket);
    };
    return { handleAddToCart };
  },
  computed: { ...mapGetters(["allTickets"]) },
});
</script>
<style scoped>
.hero {
  width: 80%;
  margin: 0 auto;
  background: #f3f4f6;
  height: 300px;
  position: relative;
  overflow: hidden;
  border-bottom-right-radius: 20px;
  border-bottom-left-radius: 20px;
  padding: 10px;
  box-shadow: 0px 4px 3px 3px #f7f7f7;
}
.hero::after {
  position: absolute;
  width: 100%;
  height: 100%;
  content: "";
  top: 0;
  background: #0f172a !important;
  z-index: 2;
  clip-path: polygon(100% 100%, 100% 0%, 40% 60%, 0% 100%);
}
h2 {
  font-size: 2.5rem;
  margin-top: 15px;
  font-weight: bold;
}
.hero p {
  margin: 10px 0;
  font-size: 1.5rem;
}
.tickets_wrapper {
  width: 80%;
  margin: 2rem auto;
  display: flex;
  gap: 2rem;
  flex-wrap: wrap;
}
.card {
  width: 200px;
  border: 1px solid whitesmoke;
  border-radius: 5px;
  padding: 5px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  box-shadow: 1px 1px 5px 2px whitesmoke;
}
.card img {
  object-fit: contain;
}
.card h3 {
  font-weight: bold;
  font-size: 1.5rem;
  text-transform: capitalize;
}
.card p {
  font-size: 1rem;
  margin: 5px 0;
  text-transform: lowercase;
}
.card .sub {
  display: flex;
  justify-content: space-between;
}
.card .sub span {
  font-size: 15px;
}
.card .sub span:nth-child(1) {
  font-weight: bold;
}
.card button {
  background: #0f172a;
  color: white;
  padding: 10px;
  width: 100%;
  position: relative;
  overflow: hidden;
}
.card .triangle {
  position: absolute;
  width: 0;
  height: 0;
  border: 24px solid transparent;
  border-top: 0;
  border-bottom: 30px solid rgb(220, 210, 210);
}
.card button span:nth-child(1) {
  left: -10px;
  transform: rotate(91deg);
}
.card button span:nth-child(2) {
  right: -10px;
  transform: rotate(269deg);
}
@media screen and (max-width: 400px) {
  .hero {
    width: 100%;
  }
  .hero h2 {
    font-size: 1.5rem;
  }
  .hero p {
    font-size: .9rem;
    width: 50%;
  }
  .tickets_wrapper{
    justify-content: center;
    width: 90%;
  }
  .card{
    width: 100%;
  }
}
</style>
