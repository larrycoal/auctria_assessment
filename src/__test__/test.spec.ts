import { mount } from "@vue/test-utils";
import {  expect, test } from "vitest";
import {createStore} from "vuex";
import App from "../App.vue";
import Home from "../Views/Home.vue";
import Cart from "../Views/Cart.vue";
import Admin from "../Views/Admi.vue";
import eventType from "../types/event";
interface stateType {
  tickets: eventType[];
  cart: eventType[];
}
let state: stateType;
  //let getters;
   let tickets: eventType[] = [
     {
       ticketName: "Festival",
       description:
         "Get your ticket for the afro vibes festival happening in summer '24 in kitchener. you dont want to miss out on this festival",
       price: 50,
       isVip: false,
       count: 100,
     },
   ];
   let cart: eventType[] = [];
   state = {
     tickets,
     cart,
   };
  const store = createStore({
    state,
    mutations: {
      addTicket(state, newstate) {
        state.tickets.push(newstate);
      },
      removeTicket(state, idx) {
        state.tickets.splice(idx, 1);
      },
      addToCart(state, newCart) {
        state.cart.push(newCart);
      },
    },
    getters: {
      allTickets: (state) => state.tickets,
      allCart: (state) => state.cart,
    },
  });
test("Application wrapper renders correctly", () => {
  const wrapper = mount(App, {
    global: {
      plugins: [store],
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
});
test("Home page renders correctly", () => {
  const wrapper = mount(Home, {
    global: {
      plugins: [store],
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
});
test("Cart page renders correctly", () => {
  const wrapper = mount(Cart, {
    global: {
      plugins: [store],
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
});
test("Admin page renders correctly", () => {
  const wrapper = mount(Admin, {
    global: {
      plugins: [store],
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
});
test("Add to cart works", async() => {
  const wrapper = mount(Home, {
    global: {
      plugins: [store],
    },
  });
  await wrapper.find(".add_to_cart").trigger('click')
  expect(store.state.cart.length).toBeGreaterThan(0);
});
