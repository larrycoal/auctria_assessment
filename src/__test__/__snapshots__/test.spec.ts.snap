// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Admin page renders correctly 1`] = `
"<div data-v-8684b5a5="" class="w-8/12 mx-auto bg-white my-4 p-4 admin_wrapper">
  <h2 data-v-8684b5a5="" class="text-3xl text-center">Add New Ticket</h2>
  <form data-v-8684b5a5="" class="flex flex-col gap-1 my-4 p-4">
    <div data-v-8684b5a5=""><label data-v-8684b5a5="">Ticket Name</label><input data-v-8684b5a5="" type="text" placeholder="Enter ticket name" required=""></div>
    <div data-v-8684b5a5=""><label data-v-8684b5a5="">Description</label><input data-v-8684b5a5="" type="text" placeholder="Enter ticket name" required=""></div>
    <div data-v-8684b5a5=""><label data-v-8684b5a5="">Count</label><input data-v-8684b5a5="" type="number" placeholder="Enter number of ticket" required=""></div>
    <div data-v-8684b5a5=""><label data-v-8684b5a5="">Price</label><input data-v-8684b5a5="" type="number" placeholder="Enter ticket price" required=""></div>
    <div data-v-8684b5a5="" class="flex flex-row my-2"><input data-v-8684b5a5="" type="checkbox"><label data-v-8684b5a5="">Vip Ticket</label></div><button data-v-8684b5a5="" class="bg-slate-900 text-gray-50 p-2 rounded">Add Ticket</button>
  </form>
</div>
<table data-v-8684b5a5="">
  <tr data-v-8684b5a5="">
    <th data-v-8684b5a5="">Name</th>
    <th data-v-8684b5a5="">description</th>
    <th data-v-8684b5a5="">price</th>
    <th data-v-8684b5a5=""></th>
  </tr>
  <tr data-v-8684b5a5="">
    <td data-v-8684b5a5="">Festival</td>
    <td data-v-8684b5a5="">Get your ticket for the afro vibes festival happening in summer '24 in kitchener. you dont want to miss out on this festival</td>
    <td data-v-8684b5a5="">50cad</td>
    <td data-v-8684b5a5=""><i data-v-8684b5a5="" class="fa-solid fa-trash cursor-pointer"></i></td>
  </tr>
</table>"
`;

exports[`Application wrapper renders correctly 1`] = `
"<header data-v-f13b4d11="" class="flex p-2 text-xl text-center bg-slate-900 text-gray-50 nav">
  <h2 data-v-f13b4d11="" class="mx-2 flex flex-1 items-start font-bold">Ticketster</h2>
  <nav data-v-f13b4d11="" class="desktop_nav">
    <router-link data-v-f13b4d11="" class="mx-4" to="/"><i data-v-f13b4d11="" class="fa-solid fa-house" style="color: rgb(255, 255, 255);"></i>Home</router-link>
    <router-link data-v-f13b4d11="" class="mx-4" to="/admin"><i data-v-f13b4d11="" class="fa-solid fa-screwdriver-wrench" style="color: rgb(255, 255, 255);"></i>Admin</router-link>
    <router-link data-v-f13b4d11="" class="mx-4 position:relative" to="/cart"><i data-v-f13b4d11="" class="fa-solid fa-cart-shopping" style="color: rgb(255, 255, 255);"></i>Cart <span data-v-f13b4d11="">0</span></router-link>
  </nav>
  <div data-v-f13b4d11="" class="hamburger"><i data-v-f13b4d11="" class="fa-solid fa-bars" style="color: rgb(247, 247, 248);"></i></div>
  <!--v-if-->
</header>
<router-view data-v-f13b4d11=""></router-view>"
`;

exports[`Cart page renders correctly 1`] = `
"<h2 data-v-846bc44f="">Your Cart</h2>
<table data-v-846bc44f="">
  <tr data-v-846bc44f="">
    <th data-v-846bc44f="">Item</th>
    <th data-v-846bc44f="">Name</th>
    <th data-v-846bc44f="">Description</th>
    <th data-v-846bc44f="">Price</th>
  </tr>
</table>
<div data-v-846bc44f="" class="cart_bottom">
  <div data-v-846bc44f="" class="container">
    <p data-v-846bc44f=""><span data-v-846bc44f="">Amount:</span><span data-v-846bc44f="">0cad</span></p>
    <p data-v-846bc44f=""><span data-v-846bc44f="">HST:</span><span data-v-846bc44f="">10%</span></p>
    <p data-v-846bc44f=""><span data-v-846bc44f="">Total:</span><span data-v-846bc44f="">0cad</span></p>
  </div>
</div>"
`;

exports[`Home page renders correctly 1`] = `
"<div data-v-fdbf00eb="">
  <div data-v-fdbf00eb="" class="hero">
    <h2 data-v-fdbf00eb="">Welcome To Ticketster</h2>
    <p data-v-fdbf00eb="">Your one stop shop to all your ticket needs</p>
  </div>
  <h2 data-v-fdbf00eb="" class="text-center text-4xl">Buy Tickets</h2>
  <ul data-v-fdbf00eb="" class="tickets_wrapper">
    <li data-v-fdbf00eb="" class="card">
      <div data-v-fdbf00eb=""><img data-v-fdbf00eb="" src="https://res.cloudinary.com/dbdrtuscd/image/upload/v1701885193/Screenshot_2023-12-06_at_12.51.35_k2pgtu.png" alt="icon for a ticket"></div>
      <!--v-if-->
      <h3 data-v-fdbf00eb="">Festival</h3>
      <p data-v-fdbf00eb="">Get your ticket for the afro vibes festival happening in summer '24 in kitchener. you dont want to miss out on this festival</p>
      <p data-v-fdbf00eb="" class="sub"><span data-v-fdbf00eb="">50cad</span><span data-v-fdbf00eb="">Available:100</span></p>
      <!--v-if--><button data-v-fdbf00eb="" class="add_to_cart"><span data-v-fdbf00eb="" class="triangle"></span>Regular Ticket<span data-v-fdbf00eb="" class="triangle"></span></button>
    </li>
  </ul>
</div>"
`;
